lib/BuildSystem/ExternalCommand.cpp

```diff
@@ -223,10 +219,11 @@ void ExternalCommand::start(BuildSystemCommandInterface& bsci,
 
 void ExternalCommand::providePriorValue(BuildSystemCommandInterface&,
                                         core::Task*,
-                                        const BuildValue& value) {
+                                        const BuildValue& value,
+                                        const basic::CommandSignature& signature) {
   if (value.isSuccessfulCommand()) {
     hasPriorResult = true;
-    priorResultCommandSignature = value.getCommandSignature();
+    priorResultCommandSignature = signature;
   }
```

Comment:
Question: If a commands signature is changed, should it really receive the priorValue? I think the answer may be no, and if it is no, then the `signature` argument to `providePriorValue` can go away (because this is the only place we use it, and we basically use it because this wasn't modeled where we now think it should be modeled)

Reply:
Resolved this by only providing the prior value to the rule if the signature has not changed, as discussed in person.